### ISSUE TITLE
Auto-generate comments

### ISSUE THREAD

**Issue by erikvullings** [TypeStrong/atom-typescript#455]
I've used Docblockr (https://atom.io/packages/docblockr) before to generate smart comments. It used to work with TypeScript files too, but recently, it doesn't anymore. However, it would be great if we could either use it again, or even better, if the comment snippet in atom-typescript would be enhanced in order to generate similar comments. 

![Example: ](https://raw.githubusercontent.com/NikhilKalige/docblockr/master/resources/function-template.gif)

**Comment by basarat** [comment]
> It used to work with TypeScript files too

It worked because the grammar tokenization for `.ts` matched what was there fore `.js` and docblockr used that. The fix would be to have the tokenization for `function` to match that in `.js` (perhaps even dropping the `.ts` suffix. 

Perhaps @Cu3PO42 can take a look.

**Comment by Cu3PO42** [comment]
I'll definitely take a look here. Two preliminary observations, though:

1. Functions are still matched as `meta.function.js` (see [grammar](https://github.com/TypeStrong/atom-typescript/blob/master/grammars/typescript.cson#L1132)).
2. docblockr does have explicit support for TypeScript (i.e. it has a definition file for it), so whatever the problem turns out to be, that might also be something one could possibly work with.

**Comment by Cu3PO42** [comment]
I just installed docblockr, and it just works for me, i.e. I get exactly the behavior shown in your screenshot:

![typescript-docblockr](https://cloud.githubusercontent.com/assets/5256300/8633561/7a0b325c-27ce-11e5-9b45-00141d5149ef.gif)

I don't know why all these exceptions pop up, but I believe that's related to the latest typescript update...  Should be unrelated here.

Are your atom-typescript and docblockr plugins up to date, @erikvullings?

**Comment by erikvullings** [comment]
Hmmm, in atom-typescript 5.0.23 it now also works for me again... Although I'm quite happy about it, it's also strange, as I've tested it with the latest versions too (atom 1.0.2), and in atom-typescript 5.0.19 it did not work yet for me. Maybe something has been fixed, or a clean install did the trick. 

Anyways, thanks a lot for checking it out and sorry to bother you! I will close this issue then.

BTW the @param statements do not show up in the atom-typescript helper tooltip. Maybe that's easy to add too?

![image](https://cloud.githubusercontent.com/assets/3140667/8634312/a632f402-27f4-11e5-8ad4-66b45a9ea239.png)
